src/menu/menu.js

```diff
@@ -15,476 +15,401 @@
  * limitations under the License.
  */
 
-(function() {
-  'use strict';
-
+/**
+ * The MaterialMenu class wraps a Material Design menu component.
+ *
+ * @export
+ */
+class MaterialMenu extends MaterialComponent {
   /**
-   * Class constructor for dropdown MDL component.
-   * Implements MDL component design pattern defined at:
-   * https://github.com/jasonmayes/mdl-component-design-pattern
+   * Initialize checkbox from a DOM node.
    *
-   * @constructor
-   * @param {HTMLElement} element The element that will be upgraded.
+   * @param {Element} root The element being upgraded.
    */
-  var MaterialMenu = function MaterialMenu(element) {
-    this.element_ = element;
+  constructor(root) {
+    super(root);
+
+    // Look for sub-nodes in the root's DOM.
+    this.list_ = this.root_.querySelector(`.${MaterialMenu.classes_.LIST}`);
+    if (!this.list_) {
+      throw new Error(
+          `MaterialMenu missing ${MaterialMenu.classes_.LIST} node.`);
+    }
+    // Outline is optional, given that it's a visual-only thing.
+    this.outline_ =
+        this.root_.querySelector(`.${MaterialMenu.classes_.OUTLINE}`);
+
+    // Initialize event listeners.
+    this.itemClickListener_ = this.itemClicked_.bind(this);
+    this.documentClickListener_ = this.documentClicked_.bind(this);
+    this.transitionListener_ = this.transitionEnd_.bind(this);
+
+    // Initialize variables.
+    this.closing_ = false;
+    this.animating_ = false;
+    this.anchor_ = null;
+    if (this.root_.hasAttribute('data-mdl-anchor')) {
+      this.anchor_ = document.querySelector(
```

Comment:
Polymer is what we recommend for developers that want to use Web Components technologies.